Source/WebKit/Scripts/copy-launchd-plist-and-create-symlink.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/sh
+
+if [[ "${SKIP_INSTALL}" == "YES" ]]; then
+    exit 0;
+fi
+
+if [[ -z "${LAUNCHD_PLIST_INSTALL_PATH}" ]]; then
+    exit 0;
+fi
+
+if [[ -z "${LAUNCHD_PLIST_FILE_NAME}" ]]; then
+    echo "Missing definition for LAUNCHD_PLIST_FILE_NAME. e.g. 'com.apple.webkit.thedaemon.plist'"
+    exit 1
+fi
+
+if [[ -z "${LAUNCHD_PLIST_INPUT_FILE}" ]]; then
+    echo "Missing definition for LAUNCHD_PLIST_INPUT_FILE. e.g. 'TheDaemon/Resources/com.apple.webkit.thedaemon.plist'"
+    exit 1
+fi
+
+# Update the launchd plist to point to the install location of the binary in its actual
+# location.  When using the system content path, we need to install there and create a
+# symlink if required.  Since the location can change based on the build configuration,
+# the launchd plists should contain placeholders for build settings INSTALL_PATH and
+# PRODUCT_NAME which we replace with the actual values from the current build configuration.
+
+GENERATED_LAUNCHD_PLIST_DIR="${TEMP_DIR}"
+mkdir -p "${GENERATED_LAUNCHD_PLIST_DIR}"
+
+GENERATED_LAUNCHD_PLIST_PATH="${GENERATED_LAUNCHD_PLIST_DIR}/${LAUNCHD_PLIST_FILE_NAME}"
+
+echo "Replacing placeholders in ${SRCROOT}/${LAUNCHD_PLIST_INPUT_FILE} and saving to ${GENERATED_LAUNCHD_PLIST_PATH}"
+
+sed \
+    -e "s|\${INSTALL_PATH}|${INSTALL_PATH}|" \
+    -e "s|\${PRODUCT_NAME}|${PRODUCT_NAME}|" \
+    "${SRCROOT}/${LAUNCHD_PLIST_INPUT_FILE}" > "${GENERATED_LAUNCHD_PLIST_PATH}"
+
+# Install the plist.
+
+LAUNCHD_PLIST_OUTPUT_FILE="${LAUNCHD_PLIST_INSTALL_PATH}/${LAUNCHD_PLIST_FILE_NAME}"
+DSTROOT_LAUNCHD_PLIST_OUTPUT_FILE="${DSTROOT}${LAUNCHD_PLIST_OUTPUT_FILE}"
+
+mkdir -p "${DSTROOT}${LAUNCHD_PLIST_INSTALL_PATH}"
+
+echo "Converting ${GENERATED_LAUNCHD_PLIST_PATH} to binary plist at ${DSTROOT_LAUNCHD_PLIST_OUTPUT_FILE}"
+plutil -convert binary1 -o "${DSTROOT_LAUNCHD_PLIST_OUTPUT_FILE}" "${GENERATED_LAUNCHD_PLIST_PATH}"
```

Comment:
I don't think my comment should block merge—as this approach seems fine—but IMO file transformations like this are better expressed using a [build rule](https://developer.apple.com/documentation/xcode/creating-build-rules-for-custom-file-types). That way, you can use a standard Copy Files phase and its install-path-computation logic, and the only freeform script needed is what does the plist modification (which itself could be simplified by using `envsubst` instead of `sed`).

We _do_ need a custom script phase to create symlinks, though. That said, if we moved the first half of this file into a build rule, the remainder begins to look a lot like [`create-symlink-to-altroot.sh`](https://github.com/WebKit/WebKit/blob/3b03219cf561b760a4ab7eef592f24bb9d72793f/Source/WebCore/Scripts/create-symlink-to-altroot.sh), which makes me wonder if it could be refactored to handle symlinking for files like this as well.